### PR TITLE
Fixing bug where SQLCMD vars weren't getting JSONified

### PR DESCRIPTION
**Pull request by Benjin**
Fixes https://github.com/microsoft/azuredatastudio/issues/23079

### CODE REVIEW

**Review by SakshiS-harma — APPROVED** [review]
Does it need to be updated for vscode-mssql?

**Review by kisantia — APPROVED** [review]

File: extensions/mssql/src/dacfx/dacFxService.ts [hunk 1]
```diff
@@ -55,12 +55,12 @@ export class DacFxService extends BaseService implements mssql.IDacFxService {
...
 		return this.runWithErrorHandling(contracts.DeployRequest.type, params);
 	}
 
 	public async generateDeployScript(packageFilePath: string, targetDatabaseName: string, ownerUri: string, taskExecutionMode: azdata.TaskExecutionMode, sqlCommandVariableValues?: Map<string, string>, deploymentOptions?: mssql.DeploymentOptions): Promise<mssql.DacFxResult> {
-		const params: contracts.GenerateDeployScriptParams = { packageFilePath: packageFilePath, databaseName: targetDatabaseName, sqlCommandVariableValues: sqlCommandVariableValues, deploymentOptions: deploymentOptions, ownerUri: ownerUri, taskExecutionMode: taskExecutionMode };
+		const params: contracts.GenerateDeployScriptParams = { packageFilePath: packageFilePath, databaseName: targetDatabaseName, sqlCommandVariableValues: Object.fromEntries(sqlCommandVariableValues), deploymentOptions: deploymentOptions, ownerUri: ownerUri, taskExecutionMode: taskExecutionMode };
```
Comment (kisantia) [inline extensions/mssql/src/dacfx/dacFxService.ts:63]: does `savePublishProfile()` below also need this change?

**Comment by Benjin** [comment]
@SakshiS-harma 
> Does it need to be updated for vscode-mssql?

Yup, will send that out next.

File: extensions/mssql/src/dacfx/dacFxService.ts (hunk 1 above)
Comment (Benjin) [reply extensions/mssql/src/dacfx/dacFxService.ts:63]: Gah, yes - I forgot to push that commit.

**Pull request merged**